src/util/logging.cpp

```diff
@@ -29,6 +28,20 @@ QMutex s_mutexStdErr;
 // The file handle for Mixxx's log file.
 QFile s_logfile;
 
+QLoggingCategory::CategoryFilter oldCategoryFilter = nullptr;
+
+void controllerDebugCategoryFilter(QLoggingCategory* category) {
+    // Configure controller.*.input/output category here, otherwise forward to to default filter.
+    constexpr char controllerPrefix[] = "controller.";
+    const char* categoryName = category->categoryName();
+    if (qstrncmp(categoryName, controllerPrefix, sizeof(controllerPrefix) - 1) == 0) {
+        category->setEnabled(QtDebugMsg, true);
+    } else {
+        oldCategoryFilter(category);
```

Comment:
It took me a while to understand the purpose of oldCategoryFilter, because the function pointer nature is not obvious due to the typedef

```suggestion
        // Call the default filter via function pointer.
        oldCategoryFilter(category);
```

Reply:
https://github.com/mixxxdj/mixxx/pull/4523/commits/81e5fe313ac2f1b50e24f8e1f120e2cf1a5fd0de